File: src/main/java/net/starlegacy/feature/gas/collectionfactors/WorldChanceFactor.java
package net.starlegacy.feature.gas.collectionfactors;

import org.bukkit.Location;

public class WorldChanceFactor extends RandomFactor {

    String world;

    public WorldChanceFactor(float chance, String world) {
        super(chance);
        this.world = world;
    }

    @Override
    public boolean factor(Location location) {
        return super.factor(location) && location.getWorld().getName().equalsIgnoreCase(world);
    }
}

File: server/src/main/kotlin/net/horizonsend/ion/server/features/gas/collectionfactors/WorldChanceFactor.kt
package net.horizonsend.ion.server.features.gas.collectionfactors

import org.bukkit.Location
import kotlin.random.Random

class WorldChanceFactor(val worlds: Map<String, Double>) : CollectionFactor() {
    override fun factor(location: Location): Boolean {
		val sample = Random.nextFloat()

		return worlds.any { (world, chance) ->
			location.world.name.equals(world, ignoreCase = true) && sample >= chance
		}
    }

	override fun canBeFound(location: Location): Boolean = worlds.any {
		(world, _) -> location.world.name.equals(world, ignoreCase = true)
	}
}
